framework/src/utils/PetscDMMoose.C

```diff
@@ -966,16 +966,16 @@ DMCreateFieldDecomposition_Moose(
           ierr = ISGetLocalSize(dmm->_embedding, &len);
           CHKERRQ(ierr);
 
-          ierr = MPI_Scan(&len,
-                          &off,
-                          1,
+          ierr = cast_int<PetscErrorCode>(MPI_Scan(&len,
```

Comment:
Wait, what are we doing here?  MPI_Scan doesn't return a PETSc error code.

Reply:
To be sure, the MPI standard mandates `MPI_SUCCESS==0`, so unless the scan actually fails we're fine, but if we see a failure the error message is going to be seriously confusing.